bcgov/HOUS-permit-portal
# Feature/bphh 760/edit requirement block elective fields

## Description
- Adds ability of super admins to mark fields as elective
- Adds tag to show input type on hover
- Adds tag to display field as elective
- Fixes bug with the state of "optional for submitter" not saving on page reload
- - Refactors `RequirementFieldEdit and `RequirementFieldDisplay` components to use reusable components to reduce duplication


## What type of PR is this? (check all applicable)

- [ X] 🍕 Feature

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-760


## Steps to QA
- Login as super admin
- Navigate to requirement block page
- Click on any to edit
- Add a requirement field if one does now already exist
- Verify you can see the input type tag
- Click Edit button and mark the checkbox to denote the field as elective
- Click done
- The elective tag should be visible
- Refresh page to confirm changes persisted
